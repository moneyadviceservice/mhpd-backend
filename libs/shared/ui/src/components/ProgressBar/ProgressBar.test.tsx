import React from 'react';
import { render, screen } from '@testing-library/react';
import { ProgressBar, VariantType } from '.';

describe('ProgressBar component', () => {
  it('renders the default correctly', () => {
    render(<ProgressBar value={30} max={100} label="30% complete" />);
    const container = screen.getByTestId('progress');
    expect(container).toMatchSnapshot();
  });

  it('renders the blue variant correctly', () => {
    render(
      <ProgressBar
        value={30}
        max={100}
        label="30% complete"
        testId="progress-component"
        variant={VariantType.BLUE}
      />,
    );
    const container = screen.getByTestId('progress-component');
    expect(container).toMatchSnapshot();
  });
});
