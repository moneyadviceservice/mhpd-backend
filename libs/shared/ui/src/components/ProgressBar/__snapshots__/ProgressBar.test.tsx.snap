// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component renders the blue variant correctly 1`] = `
<div
  class=""
  data-testid="progress-component"
>
  <label
    class="block text-center text-blue-800 font-bold text-4xl mb-6"
    for="progress"
  >
    30% complete
  </label>
  <progress
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="w-full border-2 bg-white border-slate-400 rounded-full h-[18px] shadow-bottom-slate overflow-hidden [&::-webkit-progress-bar]:bg-white [&::-webkit-progress-value]:rounded-full [&::-moz-progress-bar]:rounded-full [&::-webkit-progress-value]:bg-blue-800 [&::-moz-progress-bar]:bg-blue-800"
    id="progress"
    max="100"
    value="30"
  />
</div>
`;

exports[`ProgressBar component renders the default correctly 1`] = `
<div
  class=""
  data-testid="progress"
>
  <label
    class="block text-center text-blue-800 font-bold text-4xl mb-6"
    for="progress"
  >
    30% complete
  </label>
  <progress
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="w-full border-2 bg-white border-slate-400 rounded-full h-[18px] shadow-bottom-slate overflow-hidden [&::-webkit-progress-bar]:bg-white [&::-webkit-progress-value]:rounded-full [&::-moz-progress-bar]:rounded-full [&::-webkit-progress-value]:bg-pink-600 [&::-moz-progress-bar]:bg-pink-600"
    id="progress"
    max="100"
    value="30"
  />
</div>
`;
